src/middlewares/routeRequiresRoles.ts

```diff
@@ -0,0 +1,60 @@
+import * as debug from 'debug';
+import { NextFunction, Request, Response } from 'express';
+import * as Errors from '../server/model/errors';
+
+const debuggerInstance = debug('typescript-rest:middlewares:routeRequiresRoles');
+
+/**
+ * Middleware to check if the user has the required roles to access a route.
+ *
+ * @param authenticator extracts roles from the request.
+ * @param requiredRoles can be a single role or an array of roles. At least one role must be specified. If at least one of the roles matches the user's roles, access is granted.
+ * @returns the middleware function that checks if the user has the required roles.
+ */
+export function routeRequiresRoles(
+    authenticator: { getRoles: (req: Request, res: Response) => Array<string> },
+    requiredRoles: Array<string> | string
+) {
```

Comment:
What do you think about:

```suggestion
export function routeRequiresRole(
    authenticator: { getRoles: (req: Request, res: Response) => Array<string> },
    requiresOneOfRoles: Array<string> | string
) {
```

Reply:
TBH, I don't like it..